cdk/BackendStack.ts

```diff
@@ -43,5 +52,158 @@ export class BackendStack extends Stack {
+				environment: {
+					CACHE_TABLE_NAME: simDetailsCacheTable.tableName,
+					SIM_DETAILS_JOBS_QUEUE: resolutionJobsQueue.queueUrl,
+				},
+				initialPolicy: [
+					new IAM.PolicyStatement({
+						actions: ['dynamodb:GetItem', 'dynamodb:Query'],
+						resources: [simDetailsCacheTable.tableArn],
+					}),
+					new IAM.PolicyStatement({
+						resources: [resolutionJobsQueue.queueArn],
+						actions: ['sqs:SendMessage'],
+					}),
+				],
+			},
+		)
+
+		const getAllUsedSims = new PackedLambdaFn(
+			this,
+			'getAllUsedSims',
+			lambdaSources.getAllUsedSims,
+			{
+				layers: [baseLayer],
+				environment: {
+					CACHE_TABLE_NAME: simDetailsCacheTable.tableName,
+					APIKEY_ONOMONDO: SSM.StringParameter.valueForStringParameter(
+						this,
+						'/sim-details/onomondoKey',
+					),
+					SIM_DETAILS_JOBS_QUEUE: resolutionJobsQueue.queueUrl,
+				},
+				initialPolicy: [
+					new IAM.PolicyStatement({
+						resources: [resolutionJobsQueue.queueArn],
+						actions: ['sqs:SendMessage'],
+					}),
+				],
+			},
+		)
+		const rule = new Events.Rule(this, 'InvokeActivitiesRule', {
+			schedule: Events.Schedule.expression('rate(1 hour)'),
+			description: `Invoke the lambda that fetches usage for all active SIMs`,
+			enabled: true,
+			targets: [new EventsTargets.LambdaFunction(getAllUsedSims.fn)],
+		})
+		getAllUsedSims.fn.addPermission('InvokeByEvents', {
+			principal: new IAM.ServicePrincipal('events.amazonaws.com') as IPrincipal,
+			sourceArn: rule.ruleArn,
+		})
+
+		const storeSimInformation = new PackedLambdaFn(
+			this,
+			'storeSimInformation',
+			lambdaSources.storeSimInformation,
+			{
+				layers: [baseLayer],
+				//
+				reservedConcurrentExecutions: 10,
+				environment: {
+					CACHE_TABLE_NAME: simDetailsCacheTable.tableName,
+					APIKEY_ONOMONDO: SSM.StringParameter.valueForStringParameter(
+						this,
+						'/sim-details/onomondoKey',
+					),
+				},
+				initialPolicy: [
+					new IAM.PolicyStatement({
+						actions: ['dynamodb:PutItem'],
+						resources: [simDetailsCacheTable.tableArn],
+					}),
+				],
+			},
+		)
+
+		const api = new apigw.LambdaRestApi(this, 'simDetailsAPI', {
+			handler: getBasicSIMInformation.fn,
+			proxy: false,
+		})
+		const iccidVar = api.root.addResource('iccid')
+		iccidVar.addMethod('GET')
+
+		const iccid = iccidVar.addResource('{iccid}')
+		iccid.addMethod('GET')
+
+		storeSimInformation.fn.addPermission('invokeBySQS', {
+			principal: new IAM.ServicePrincipal('sqs.amazonaws.com'),
+			sourceArn: resolutionJobsQueue.queueArn,
+		})
+
+		new Lambda.EventSourceMapping(this, 'invokeLambdaFromNotificationQueue', {
+			eventSourceArn: resolutionJobsQueue.queueArn,
+			target: storeSimInformation.fn,
+			batchSize: 10,
+		})
+
+		resolutionJobsQueue.grantConsumeMessages(storeSimInformation.fn)
+		resolutionJobsQueue.grantConsumeMessages(getAllUsedSims.fn)
+		new cdk.CfnOutput(this, 'simDetailsApi', {
+			exportName: `${this.stackName}:simDetailsApi`,
+			value: api.url,
```

Comment:
```suggestion
		new cdk.CfnOutput(this, 'simDetailsApiURL', {
			exportName: `${this.stackName}:simDetailsApiURL`,
			value: api.url,
```

Make it explicit what this variable contains.